File: component/src/main/java/com/mackwu/component/fragment/LifecycleFragment.java
package com.mackwu.component.fragment;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mackwu.component.base.BaseFragment;

/**
 * ===================================================
 * Created by MackWu on 2020/6/20 1:36
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public abstract class LifecycleFragment extends BaseFragment {

    private static final String HEAD = LifecycleFragment.class.getSimpleName() + " ";

    @Override
    public void onAttach(@NonNull final Context context) {
        super.onAttach(context);
        Log.d("TAG", HEAD + "onAttach...");
    }

    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("TAG", HEAD + "onCreate...");
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull final LayoutInflater inflater, @Nullable final ViewGroup container, @Nullable final Bundle savedInstanceState) {
        Log.d("TAG", HEAD + "onCreateView...");
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Log.d("TAG", HEAD + "onViewCreated...");
    }

    @Override
    public void onActivityCreated(@Nullable final Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        Log.d("TAG", HEAD + "onActivityCreated...");
    }

    @Override
    public void onViewStateRestored(@Nullable final Bundle savedInstanceState) {
        super.onViewStateRestored(savedInstanceState);
        Log.d("TAG", HEAD + "onViewStateRestored...");
    }

    @Override
    public void onStart() {
        super.onStart();
        Log.d("TAG", HEAD + "onStart...");
    }

    @Override
    public void onResume() {
        super.onResume();
        Log.d("TAG", HEAD + "onResume...");
    }

    @Override
    public void onPause() {
        super.onPause();
        Log.d("TAG", HEAD + "onPause...");
    }

    @Override
    public void onStop() {
        super.onStop();
        Log.d("TAG", HEAD + "onStop...");
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.d("TAG", HEAD + "onDestroyView...");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d("TAG", HEAD + "onDestroy...");
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.d("TAG", HEAD + "onDetach...");
    }

    @Override
    public void onHiddenChanged(final boolean hidden) {
        super.onHiddenChanged(hidden);
        Log.d("TAG", HEAD + "onHiddenChanged...");
    }
}

File: fragment/src/main/java/com/mackwu/fragment/base/LifecycleFragment.kt
package com.mackwu.fragment.base

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.mackwu.fragment.R

/**
 * ===================================================
 * Created by MackWu on 2019/9/30 11:46
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 * 生命周期
 *
 * replace Fragment
 * 09-30 01:20:31.777 4020-4020/com.mackwu.fragment D/TAG: onAttach...
 * 09-30 01:20:31.777 4020-4020/com.mackwu.fragment D/TAG: onCreate...
 * 09-30 01:20:31.778 4020-4020/com.mackwu.fragment D/TAG: onCreateView...
 * 09-30 01:20:31.779 4020-4020/com.mackwu.fragment D/TAG: onActivityCreated...
 * 09-30 01:20:31.779 4020-4020/com.mackwu.fragment D/TAG: onStart...
 * 09-30 01:20:31.779 4020-4020/com.mackwu.fragment D/TAG: onResume...
 * 按返回键
 * 09-30 01:21:24.224 4020-4020/com.mackwu.fragment D/TAG: onPause...
 * 09-30 01:21:24.228 4020-4020/com.mackwu.fragment D/TAG: onStop...
 * 09-30 01:21:24.228 4020-4020/com.mackwu.fragment D/TAG: onDestroyView...
 * 09-30 01:21:24.229 4020-4020/com.mackwu.fragment D/TAG: onDestroy...
 * 09-30 01:21:24.229 4020-4020/com.mackwu.fragment D/TAG: onDetach...
 */
class LifecycleFragment : Fragment() {

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        Log.d("TAG", "onAttach...")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("TAG", "onCreate...")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.d("TAG", "onCreateView...")
        return inflater.inflate(R.layout.fragment_home, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        Log.d("TAG", "onActivityCreated...")
    }

    override fun onStart() {
        super.onStart()
        Log.d("TAG", "onStart...")
    }

    override fun onResume() {
        super.onResume()
        Log.d("TAG", "onResume...")
    }

    override fun onPause() {
        super.onPause()
        Log.d("TAG", "onPause...")
    }

    override fun onStop() {
        super.onStop()
        Log.d("TAG", "onStop...")
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.d("TAG", "onDestroyView...")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("TAG", "onDestroy...")
    }

    override fun onDetach() {
        super.onDetach()
        Log.d("TAG", "onDetach...")
    }
}